### DISCUSSION TITLE
error 1033 argo tunnel cloudflare

### DISCUSSION THREAD

**Discussion by DigoGibin** [indomus/forum#2442] [Networking]
Ciao a tutti, ho seguito la guida sul forum per poter raggiungere la mia istanza home assistant da remoto tramite cloudflare stanziato in container docker.
Una volta terminato mi trovo questo errore ( Error 1033 Argo Tunnel error) quando cerco di connettermi ad home assistant.
Da quanto ho letto dovrebbe essere un problema di DNS ma essendo negato in materia non capisco dove guardare per risolvere il problema. 
Avete idee per capire di cosa si tratta ? 
allego  dei dettagli e qualche immagine sperando in un aiuto

- uso un dominio di aruba: G********.it al quale ho sostituito i server con quelli di clodflare consigliati
![server cloudflare](https://user-images.githubusercontent.com/88708826/234005331-c4af0172-24cf-4ae2-903f-80619d7a6260.png)

- unico record dns presente al termine della guida
![record dns](https://user-images.githubusercontent.com/88708826/234006793-86722f1a-07f4-4529-94eb-b46ab044b355.png)

- log container cloudlfared
![log cloudflared](https://user-images.githubusercontent.com/88708826/234007231-05c34fed-079c-4641-a5f4-39a893912bf6.png)

- check tunnel
![tunnel cloudflared](https://user-images.githubusercontent.com/88708826/234007703-6cb5883d-2e2a-47d6-8441-f7435851cdba.png)

- indirizzo di rete
![rete](https://user-images.githubusercontent.com/88708826/234009532-e95f6550-ffd7-4db8-85cb-0928c4685730.png)

- configurazione home assistant
![configurazione](https://user-images.githubusercontent.com/88708826/234009853-57473ced-ee59-47d2-8002-d16525c4ab84.png)

Ringrazio anticipatamente chi abbia voglia di aiutarmi.
Diego

**Comment by DigoGibin** [comment]
Ciao a tutti volevo avvisare che ho risolto eseguendo la procedura dalla dashboard di cloudflare stanziando su docker la loro configurazione del container

Grazie

**Reply by snakuzzo** [reply]
La nostra guida si basa proprio sulla guida ufficiale, cosa hai fatto di diverso?

**Comment by DigoGibin** [comment]
Ciao,
per quanto i server e configurazione lato homeassistant ho seguito le vostre guide è per la creazione e configuraizone del container che ho avuto problemi. Seguendo le indicazioni dal sito cloudlfare (non so se posso mettere il link, spero di si altrimenti cancello)
https://developers.cloudflare.com/cloudflare-one/connections/connect-apps/install-and-setup/tunnel-guide/remote/
la grossa differenza che ho potuto notare e il metodo di creazione del tunnel e la semplicità di creare il container con solo un comando già tutto configurato in automatico da loro.

necessita la registrazione al loro prodotto zerotrust ma riamane sempre un piano free.

ciao

**Comment by Domino85** [comment]
Ciao!
Io sto avendo lo stesso problema. Mentre il tunnel generato in locale grazie alla vostra guida rimane inattivo e sembra (dai log) iniziarsi continuamente, il metodo descritto da DigoGibin funziona e riesco ad accedere a HA remotamente. Il problema però è che con il comando di Cloudflare (con il token) non riesco più a fare nulla sul raspberry perché il terminale continua a mostrare i log in tempo reale del tunnel. C'è una soluzione? DigoGibin come hai risolto?
Grazie!

**Reply by DigoGibin** [reply]
Ciao, non capisco cosa intendi con il comando di cloudflare per connetterti al raspberry?

**Comment by gitmed1** [comment]
Ciao, stesso identico problema. Ora provo la tua soluzione

**Reply by gitmed1** [reply]
Ho scoperto che il mio tunnel creato con la guida indomus risulta inattivo ed è per questo che probabilmente non funziona ma non capisco quale comando dargli per attivarlo... Poi ho provato a crea un nuovo tunnel... Che è in funzione è healty ma non capisco come farlo arrivare alla mia istanza home assistant... Qualche buon anima sa aiutarmi? Grazie mille in anticipo

**Reply by snakuzzo** [reply]
se hai creato un nuovo tunnel, vuol dire che hai creato anche un altro CNAME (che sarà quello da utilizzare per raggiungere HA). Controlla sulla dashboard di Cloudflare che hai combinato

**Reply by gitmed1** [reply]
Avevo creato un tunnel con zerotrust come da guida indicata in link da digogibin, ma non avevo creato nessun cname, oro l'ho eliminato perchè voglio sistemarlo. sulla dashboard ora c'è il mio sito con scritto attivo, allego schermate:

docker logs cloudflared
cloudflared version 2024.5.0 (built 2024-05-16-1623 UTC)
cloudflared version 2024.5.0 (built 2024-05-16-1623 UTC)
cloudflared version 2024.5.0 (built 2024-05-16-1623 UTC)
cloudflared version 2024.5.0 (built 2024-05-16-1623 UTC)
cloudflared version 2024.5.0 (built 2024-05-16-1623 UTC)
cloudflared version 2024.5.0 (built 2024-05-16-1623 UTC)
...continua

![Schermata del 2024-05-31 17-14-42](https://github.com/indomus/forum/assets/160527471/66066a9d-2e05-4770-9c16-9668de456f8c)

![Schermata del 2024-05-31 16-56-34](https://github.com/indomus/forum/assets/160527471/7eaabb10-cd70-4de0-b5e1-262e997b6de5)
![Schermata del 2024-05-31 17-15-31](https://github.com/indomus/forum/assets/160527471/eececd01-cdd6-4d58-b62a-371cfd58a9e5)

ecco non riesco a capire perchè mi da inattivo...e non trovo il comando per attivarlo

io utilizzo un mini pc con  linux debian 12, ho home assistant core su docker...qualche aiuto...oppure, essendo neofita di linux, qualche comando che possa aiutarvi a capire il problema? grazie in anticipo.

**Reply by snakuzzo** [reply]
Quel log che hai postato mi sembra strano...e se vedi solo quello mi sembra normale che ti dia il tunnel inattivo.
Quando avvii il container dovresti avere qualcosa del genere...

```
2024-05-17T07:48:59Z INF Starting tunnel tunnelID=xxxxxxxxxxx
2024-05-17T07:48:59Z INF Version 2024.5.0
2024-05-17T07:48:59Z INF GOOS: linux, GOVersion: go1.22.2-devel-cf, GoArch: amd64
2024-05-17T07:48:59Z INF Settings: map[cred-file:/home/nonroot/.cloudflared/xxxxxxxxxxx.json credentials-file:/home/nonroot/.cloudflared/xxxxxxxxxxx.json no-autoupdate:true]
2024-05-17T07:48:59Z INF Generated Connector ID: xxxxxxxxxxx
```

Per quanto riguarda il CNAME di cui si parlava, il CNAME viene creato quando esegui la procedura, non lo vai a creare tu sulla dashboard di cloudflare.

**Reply by gitmed1** [reply]
A me quella scritta che hai postato non è mai uscita, mi uscivano le risposte esattamente come si vede nella guida "https://indomus.it/guide/collegarsi-da-remoto-a-home-assistant-via-reti-provider-tradizionali-o-che-usino-nat-via-cloudflare/" che comando hai utilizzato per avviare il container? Io ho usato quello nella guida : docker run -d --name="cloudflared" -v "~/cloudflared:/home/nonroot/.cloudflared" --restart always cloudflare/cloudflared:latest

Grazie per la pazienza, ma ho poca conoscenza di linux

**Reply by snakuzzo** [reply]
Ovvio...quello che ti ho postato io è quello che vedo nei log quando avvio questo container. Ovvio che il messaggio non è come quello della guida... perché quello è il log nel momento in cui CREI il tunnel.

Io uso docker compose, ma il punto non è questo. Avviarlo con docker o con compose non cambia nulla. I comandi in guida sono stati testati e ritestati...quelli sono

**Reply by snakuzzo** [reply]
Però pensavo una cosa...se il log è quello della guida, perché prima hai postato una cosa diversa?
Se il log fosse quello della guida, il tunnel dovrebbe essere correttamente funzionante

**Reply by gitmed1** [reply]
Il log che ho postato mi esce col comando "docker logs cloudflared" , mentre per crearlo ho seguito la guida alla lettera... Ho due domini e ho seguito la guida per 2 volte con lo stesso errore... Quello indicato dall'utente che ha aperto il post... Ma il CNAME creato deve esserci anche sul register o basta che ci sia su Cloudflare? Altra cosa, il tunnel come vedi dall'immagine sopra è inattivo, non c'è un comando per forzare l'attivazione? Oppure, potrebbe essere che bisogna aprire qualche porta sul router per far funzionare il container Cloudflare? Non so più che fare

**Reply by snakuzzo** [reply]
Il cane lo vedi solo su Cloudflare
Non c'è alcun comando, il comando è quello del container quando viene avviato. È il container di cloudflared che stabilisce la connessione con Cloudflare, quindi non devi aprire alcuna porta.
Poi ripeto...quel log che hai postato no va bene.

Riavvia il container e posta tutto il log

**Reply by gitmed1** [reply]
~$ docker container stop cloudflared
cloudflared
:~$ docker container rm  cloudflared
cloudflared
:~$ docker run -d --name="cloudflared" -v "~/cloudflared:/home/nonroot/.cloudflared" --restart always cloudflare/cloudflared:latest
docker: Error response from daemon: create ~/cloudflared: "~/cloudflared" includes invalid characters for a local volume name, only "[a-zA-Z0-9][a-zA-Z0-9_.-]" are allowed. If you intended to pass a host directory, use absolute path.
See 'docker run --help'.
:~$ docker run -d --name="cloudflared" -v "/home/deb/cloudflared:/home/nonroot/.cloudflared" --restart always cloudflare/cloudflared:latest
59d16283f58e4e2154137920dd78b1e8534a8a455adxxxxxxxxxx
:~$ docker logs cloudflared
cloudflared version 2024.5.0 (built 2024-05-16-1623 UTC)
cloudflared version 2024.5.0 (built 2024-05-16-1623 UTC)
cloudflared version 2024.5.0 (built 2024-05-16-1623 UTC)
cloudflared version 2024.5.0 (built 2024-05-16-1623 UTC)
cloudflared version 2024.5.0 (built 2024-05-16-1623 UTC)
cloudflared version 2024.5.0 (built 2024-05-16-1623 UTC)
cloudflared version 2024.5.0 (built 2024-05-16-1623 UTC)
cloudflared version 2024.5.0 (built 2024-05-16-1623 UTC)

ecco, penso che così ho riavviato il container,. l'unico errore in tutta la guida è quell'errore che mi ha dato per avviare il container ma poi risolto indicandolgi l'intero path...ma i log come vedi sono uguali.

**Reply by gitmed1** [reply]
se eseguo docker ps:
xxxxxxxx   cloudflare/cloudflared:latest                  "cloudflared --no-au…"   13 minutes ago   Restarting (0) 23 seconds ago                                                                                                                 cloudflared

**Reply by snakuzzo** [reply]
Se leggi bene i log...

```
Error response from daemon: create /cloudflared: "/cloudflared" includes invalid characters for a local volume name, only "[a-zA-Z0-9][a-zA-Z0-9_.-]" are allowed. If you intended to pass a host directory, use absolute path.
```

C'è scritto che quando crei il container e gli passi dei volumi dovresti passargli il path assoluto. Tu invece hai usato la tilde che è uno short character per indicare la home dell'utente...e quindi il volume non viene montato...e quindi tutta la configurazione non la vede.

Altro consiglio...usa docker compose

**Reply by gitmed1** [reply]
Forse non leggi le righe dopo perché forse il sistema del forum me le ha cancellate ma io successivamente rilancio il comando indicando /home/miautenza/etc... È mi rilascia una stringa con diversi caratteri alfanumerici. Cosa che dovrebbe indicare che il container è stato creato...

**Reply by snakuzzo** [reply]
Ok...hai fatto un casino con la formattazione di quello che hai scritto.
Comunque, quello è l'id univoco del container che è stato creato/avviato.
Resta il fatto che quel log non va bene, quindi hai tappato qualcosa in tutto il giro, che è difficile da capire in base a questi dettagli che hai condiviso.
Mi dispiace

**Reply by snakuzzo** [reply]
E altra cosa...visto che dici che quando fai la procedura da guida e ti appare il log così come descritto in guida, mi viene da pensare che sbagli qualcosa quando dopo vai a riavviare il container negli step successivi la creazione/attivazione del tunnel

**Reply by gitmed1** [reply]
dimmi di quali dettagli hai bisogno che provo a fornirteli e per quanto riguarda il fatto di riavviare il container di che parte di guida si tratta? che provo a riperla, se no niente, mi tengo duckdns che funziona abbastanza bene.

**Comment by gitmed1** [comment]
forse ho trovato parte del problema, ma non so come risolverlo...
`cloudflared tunnel list

2024-06-01T12:21:02Z ERR Cannot determine default origin certificate path. No file cert.pem in [~/.cloudflared ~/.cloudflare-warp ~/cloudflare-warp /etc/cloudflared /usr/local/etc/cloudflared]. You need to specify the origin certificate path by specifying the origincert option in the configuration file, or set TUNNEL_ORIGIN_CERT environment variable originCertPath=
Error locating origin cert: client didn't specify origincert path
`

**Reply by snakuzzo** [reply]
Scusa, ma ora cosa c'entra questo ? Da dove hai lanciato questo comando ?

**Reply by gitmed1** [reply]
dalla shell del server

**Reply by snakuzzo** [reply]
ma cosa c'entra la shell del server se tu stai eseguendo il container docker ?
Mica dalla shell del server vedi quello che vede il container ?
Ti sfugge forse la base del funzionamento di docker

**Reply by snakuzzo** [reply]
Comunque, faccio anche io delle prove...ti aggiorno appena posso

**Reply by snakuzzo** [reply]
Credo di aver capito l'inghippo.
Per la parte docker run dovremmo correggere la guida.

Ti confermo che il seguente comando è sbagliato, in due punti...

```
docker run -d --name="cloudflared" -v "~/cloudflared:/home/nonroot/.cloudflared" --restart always cloudflare/cloudflared:latest
```

1. non vanno usati i doppi apici nel bind del volume
2. il comando è incompleto perché alla fine deve essere indicata l'opzione da passare al comando di default del container. Quindi gli va passato il comando `tunnel run` e l' `UUID` del tunnel

Ecco il comando corretto...

```
docker run -d --name="cloudflared" -v ~/cloudflared:/home/nonroot/.cloudflared --restart always cloudflare/cloudflared:latest tunnel run <UUID>
```

... dove `<UUID>` è appunto l'`UUID` che hai anche nel `config.yml`

Come avrai capito...si spiega anche il log che vedevi prima.
Di fatto veniva lanciato semplicemente il comando `cloudflared` che di fatto non fa nulla e quindi il container si fermava.
Poi, visto che hai indicato l'opzione restart, si riavviava e faceva in loop sempre la stessa cosa...ovvero niente (oltre a stampare la versione del software)

**Reply by gitmed1** [reply]
Oh grazie mille, appena posso provo, solo un'ultima curiosità, ma nel comando c'è "nonroot", equivale a mettere il nome utente o è tutta un'altra cosa? Ps: io ho usato nonroot

**Reply by gitmed1** [reply]
FUNZIONA!!!! grazie!!!

**Reply by snakuzzo** [reply]
Perfetto! 

`nonroot` è la directory all'interno del container, deve essere così perché l'immagine docker questo prevede.

**Reply by fvnfa** [reply]
scusate una cosa, ho fatto praticamete tutto presumo 10 volte... 
andando a scrivere https://casamia.com mi dà sempre : 400: Bad Request
sto diventando pazzo ( premetto che cerco di leggere per filo e per segno sempre tutti i manuali che sono scritti veramente bene)

**Reply by snakuzzo** [reply]
ti sei perso la modifica da fare nel `configuration.yaml` nel blocco `http:` dove dovrai impostare `trusted_proxies`

**Reply by gitmed1** [reply]
Ma, domanda stupida, https://casamia.com è tuo come dominio?

**Reply by fvnfa** [reply]
gitmed1 , no , è l'esempio della guida
snakuzzo prima di tutto grazie mille, poi volevo dirti che ho inserito nel c.yaml 172.17.0.0/16 come scritto nella procedura per trovare l'ip ma niet... hai qualche altro consigio da darmi?

**Reply by snakuzzo** [reply]
vai sui log di Core e verifica quale IP ti segnala nel warning quando ti dà errore HTTP 400

**Reply by fvnfa** [reply]
172.17.0.2

**Reply by snakuzzo** [reply]
Si vede che hai sbagliato il configuration.yaml, oppure ne stai modificando uno che non è quello che in realtà hai nella config di core

**Comment by fvnfa** [comment]
diciamo che mi scrive questo:
 A request from a reverse proxy was received from 172.17.0.2, but your HTTP integration is not set-up for reverse proxies

**Reply by snakuzzo** [reply]
già ti ho risposto poco sopra.

P.S. Non creare nuovi thread nella stessa discussione...rispondi sempre sotto l'ultimo messaggio. Grazie
